unime/src/lib/events/History.svelte

```diff
@@ -1,71 +1,99 @@
-      lastConnected: 'n/a',
-    },
-    title: 'Initial connection',
-    timestamp: '2023-08-03T12:23:41.218Z',
-    credentials: [],
-  };
+    let date = history.date;
+    let credentials = history.credentials;
 
-  const data_1 = {
-    ...events[0],
-    connection: {
-      domain: 'impierce.com',
-      id: 'impierce',
-      url: 'https://impierce.com',
-      lastConnected: 'n/a',
-    },
-    title: 'Data shared',
-    timestamp: '2023-08-03T12:23:42.749Z',
-    credentials: [$state.credentials[0], $state.credentials[1]],
-  };
+    switch (history.event_type) {
+      case 'CredentialsAdded': {
+        title = $LL.TIMELINE.CREDENTIALS_ADDED() + ' ' + history.issuer_name;
+        icon = DownloadSimple;
+        break;
+      }
+      case 'CredentialsShared': {
+        title = $LL.TIMELINE.CREDENTIALS_SHARED() + ' ' + history.issuer_name;
+        icon = ShareFat;
+        break;
+      }
+      case 'ConnectionAdded': {
+        title = $LL.TIMELINE.CONNECTION_ADDED() + ' ' + history.issuer_name;
+        icon = PlugsConnected;
+        break;
+      }
+    }
 
-  const data_2 = {
-    connection: {
-      id: 'iota',
-    },
-    title: 'Initial connection',
-    timestamp: '2024-01-09T11:53:53.937+00:00',
-    credentials: [$state.credentials[2]],
-  };
+    return {
+      title,
+      icon,
+      date,
+      credentials,
+    } as DisplayEvent;
+  });
 
-  let eventsList = [data_1, data_0, data_2];
+  function hasNextElement(i: number): boolean {
+    return i + 1 < events.length;
+  }
 </script>
 
-<div class="relative flex h-full flex-col">
-  {#if events.length === 0}
+<div class="relative mt-6 flex h-full flex-col">
+  {#if $state.history.length === 0}
     <div class="flex h-full flex-col items-center justify-center">
       <p class="text-[14px]/[22px] font-medium text-slate-500 dark:text-slate-300">{$LL.TIMELINE.EMPTY()}</p>
     </div>
   {:else}
-    <div class="flex grow flex-col space-y-8 pr-4 pt-4">
-      {#each eventsList as event}
-        <div class="flex justify-between">
-          <div class="z-10 mr-3 h-6 w-6 overflow-hidden rounded-full bg-white p-0.5 ring-8 ring-silver">
-            <Image id={event.connection.id} imgClass="h-full w-full" />
+    <div class="ml-2 mt-6">
+      {#each events as event, i}
+        <div class="flex flex-row">
+          <div class="mt-2 flex flex-col items-center">
+            <div class="z-10 flex items-center justify-center rounded-full bg-white ring-8 ring-white dark:ring-silver">
+              <svelte:component this={event.icon} class="h-4 w-4 " />
+            </div>
+            {#if hasNextElement(i)}
+              <!-- Vertical line on the left -->
+              <div class="mb-2 mt-4 h-full rounded-full border border-slate-200 border-y-gray-200"></div>
+            {/if}
           </div>
-          <div class="grow">
-            <HistoryEntry {...event} />
+          <div class="ml-6 mt-[-8px] flex grow justify-between pb-10">
```

Comment:
If you want to align it in the middle, you'd need `mt-[-5px]` since two lines of text are `42px` and the icon to left is `32px`, so `(42px - 32px) / 2 = 5px`.

Reply:
Using a non-absolute way of alignment would be better (using flex).